src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/AgreementProcessServiceImpl.scala

```diff
@@ -20,25 +25,44 @@ final case class AgreementProcessServiceImpl(invoker: AgreementProcessInvoker, a
   private implicit val logger: LoggerTakingImplicit[ContextFieldsToLog] =
     Logger.takingImplicit[ContextFieldsToLog](this.getClass)
 
-  private val serviceName: String = "attribute-registry"
+  private val serviceName: String     = "attribute-registry"
+  private val missingEntityId: String = "NoIdentifier"
 
-  override def createAgreement(seed: AgreementPayload)(implicit contexts: Seq[(String, String)]): Future[Agreement] = {
+  override def createAgreement(seed: AgreementPayload)(implicit contexts: Seq[(String, String)]): Future[Agreement] =
     for {
       (bearerToken, correlationId, ip) <- extractHeaders(contexts).toFuture
       request = api.createAgreement(xCorrelationId = correlationId, agreementPayload = seed, xForwardedFor = ip)(
         BearerToken(bearerToken)
       )
-      result <- invoker.invoke(request, s"Creating agreement with seed $seed", invocationRecovery())
+      result <- invoker.invoke(request, s"Creating agreement with seed $seed", invocationRecovery(None))
     } yield result
-  }
 
-  private def invocationRecovery[T]()
-    : (ContextFieldsToLog, LoggerTakingImplicit[ContextFieldsToLog], String) => PartialFunction[Throwable, Future[T]] =
+  override def getAgreementById(agreementId: UUID)(implicit contexts: Seq[(String, String)]): Future[Agreement] =
+    for {
+      (bearerToken, correlationId, ip) <- extractHeaders(contexts).toFuture
+      request = api.getAgreementById(
+        xCorrelationId = correlationId,
+        agreementId = agreementId.toString,
+        xForwardedFor = ip
+      )(BearerToken(bearerToken))
+      result <- invoker.invoke(
+        request,
+        s"Retrieving agreement $agreementId",
+        invocationRecovery(Some(agreementId.toString))
+      )
+    } yield result
+
+  private def invocationRecovery[T](
```

Comment:
I copied this handling from other services.
#tomorrow the handling with common/generic errors could become the default partial function used in the `invoke` method